pkg/kubefedctl/federate/util.go

```diff
@@ -207,7 +207,16 @@ func getResourcesInNamespace(config *rest.Config, namespace string, skipAPIResou
 		targetResources := resources{apiResource: apiResource}
 		for _, item := range resourceList.Items {
 			resource := item
-			targetResources.resources = append(targetResources.resources, &resource)
+			errors := validation.IsDNS1123Subdomain(resource.GetName())
+			if len(errors) == 0 {
+				targetResources.resources = append(targetResources.resources, &resource)
+			} else {
+				klog.Warningf("Skipping resource %s of type %s because it does not conform to the DNS-1123 subdomain spec.", resource.GetName(), apiResource.Name)
```

Comment:
To simplify this into one klog invocation, you could print all the errors using `%v`.